src/jit/gtlist.h

```diff
@@ -69,20 +69,21 @@ GTNODE(CKFINITE         , GenTreeOp          ,0,(GTK_UNOP|GTK_NOCONTAIN)) // Che
 GTNODE(LCLHEAP          , GenTreeOp          ,0,(GTK_UNOP|GTK_NOCONTAIN)) // alloca()
 GTNODE(JMP              , GenTreeVal         ,0,(GTK_LEAF|GTK_NOVALUE))   // Jump to another function
 
-GTNODE(ADDR             , GenTreeOp          ,0,GTK_UNOP)               // address of
-GTNODE(IND              , GenTreeOp          ,0,GTK_UNOP)               // load indirection
-GTNODE(STOREIND         , GenTreeStoreInd    ,0,(GTK_BINOP|GTK_NOVALUE))  // store indirection
-
-                                                                        // TODO-Cleanup: GT_ARR_BOUNDS_CHECK should be made a GTK_BINOP now that it has only two child nodes
-GTNODE(ARR_BOUNDS_CHECK , GenTreeBoundsChk   ,0,(GTK_SPECIAL|GTK_NOVALUE))// array bounds check
-GTNODE(OBJ              , GenTreeObj         ,0,(GTK_UNOP|GTK_EXOP))      // Object that MAY have gc pointers, and thus includes the relevant gc layout info.
-GTNODE(STORE_OBJ        , GenTreeBlk         ,0,(GTK_BINOP|GTK_EXOP|GTK_NOVALUE)) // Object that MAY have gc pointers, and thus includes the relevant gc layout info.
-GTNODE(BLK              , GenTreeBlk         ,0,GTK_UNOP)               // Block/object with no gc pointers, and with a known size (e.g. a struct with no gc fields)
-GTNODE(STORE_BLK        , GenTreeBlk         ,0,(GTK_BINOP|GTK_NOVALUE))  // Block/object with no gc pointers, and with a known size (e.g. a struct with no gc fields)
-GTNODE(DYN_BLK          , GenTreeBlk         ,0,GTK_SPECIAL)            // Dynamically sized block object
-GTNODE(STORE_DYN_BLK    , GenTreeBlk         ,0,(GTK_SPECIAL|GTK_NOVALUE))// Dynamically sized block object
+GTNODE(ADDR             , GenTreeOp          ,0,GTK_UNOP)                // address of
+
+GTNODE(IND              , GenTreeIndir       ,0,GTK_UNOP)                // load indirection
```

Comment:
It would be nice to have protection from such cases in the future (smth like for each GT_* opcode types in gtlist.h and gtstructs.h should match), or maybe we do not need two separate files...

Reply:
Yes but at this stage it's kind of pointless due to some opers claiming to be BINOP/UNOP when they are not - LIST (should not exist), SIMD/HWIntrinsic/LEA (should be special, not binop), RET (it's sometimes UNOP and sometimes 0-ary, not clear what to do with it) and maybe others I didn't notice. If we manage to cleanup this mess then we can probably look at improving this, including adding some kind of validation to prevent errors.